pkg/metrics/metrics.go

```diff
@@ -0,0 +1,147 @@
+package metrics
+
+import (
+	"net/http"
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/klog/v2"
+)
+
+var (
+	// global instance of MetricRecorder
+	r *MetricRecorder = NewRecorder()
+)
+
+type MetricRecorder struct {
+	registry metrics.KubeRegistry
+	register sync.Once
+
+	apiDuration   *metrics.HistogramVec
+	apiError      *metrics.CounterVec
+	apiThrottle   *metrics.CounterVec
+	internalValue *metrics.HistogramVec
+	internalError *metrics.CounterVec
+}
+
+// Recorder returns the global instance of MetricRecorder
+func Recorder() *MetricRecorder {
+	return r
+}
+
+// NewRecorder initializes a new MetricRecorder with default metric definitions.
+func NewRecorder() *MetricRecorder {
+	return &MetricRecorder{
+		registry: metrics.NewKubeRegistry(),
+		apiDuration: metrics.NewHistogramVec(
+			&metrics.HistogramOpts{
+				Name:           "cloudprovider_aws_api_request_duration_seconds",
+				Help:           "Latency of AWS API calls",
+				StabilityLevel: metrics.ALPHA,
+			},
+			[]string{"request"},
+		),
+		apiError: metrics.NewCounterVec(
+			&metrics.CounterOpts{
+				Name:           "cloudprovider_aws_api_request_errors",
+				Help:           "AWS API errors",
+				StabilityLevel: metrics.ALPHA,
+			},
+			[]string{"request"},
+		),
+		apiThrottle: metrics.NewCounterVec(
+			&metrics.CounterOpts{
+				Name:           "cloudprovider_aws_api_throttled_requests_total",
+				Help:           "AWS API throttled requests",
+				StabilityLevel: metrics.ALPHA,
+			},
+			[]string{"operation_name"},
+		),
+		internalValue: metrics.NewHistogramVec(
+			&metrics.HistogramOpts{
+				Name:           "ebs_csi_operation",
```

Comment:
Nit: Rename to "ebs_csi_operation_duration_seconds" to match above AWS API one.

Reply:
Went with `ebs_csi_aws_com` in latest rev for the prefix since that's our official driver name: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/8f246ed15e1aec91c4e5c72cda46f37c7533b5bf/pkg/driver/driver.go#L45

(`.` characters are illegal hence the `_`).

How does this look?

```
internalDurationMetric = "ebs_csi_aws_com_duration_seconds"
internalValueMetric    = "ebs_csi_aws_com_operation"
internalErrorMetric    = "ebs_csi_aws_com_errors_total"
```
Been considering `ebs_csi_aws_com_operation` -> `ebs_csi_aws_com_operation_values`.

Users will be relying on these names when scraping metrics so lets make sure we get this right before PR is merged.